diff --git a/node_modules/@shopify/react-native-skia/ios/RNSkia-iOS/RNSkMetalCanvasProvider.mm b/node_modules/@shopify/react-native-skia/ios/RNSkia-iOS/RNSkMetalCanvasProvider.mm
index 0c73eee..cefb1e3 100644
--- a/node_modules/@shopify/react-native-skia/ios/RNSkia-iOS/RNSkMetalCanvasProvider.mm
+++ b/node_modules/@shopify/react-native-skia/ios/RNSkia-iOS/RNSkMetalCanvasProvider.mm
@@ -64,7 +64,8 @@ bool RNSkMetalCanvasProvider::renderToCanvas(
   // accessed from the main thread so we need to check here.
   if ([[NSThread currentThread] isMainThread]) {
     auto state = UIApplication.sharedApplication.applicationState;
-    if (state == UIApplicationStateBackground) {
+    auto isExtensionContext = [[[NSBundle mainBundle] bundlePath] hasSuffix:@".appex"];
+    if (state == UIApplicationStateBackground && !isExtensionContext) {
       // Request a redraw in the next run loop callback
       _requestRedraw();
       // and don't draw now since it might cause errors in the metal renderer if
